chalk-solve/src/recursive.rs

```diff
@@ -138,11 +199,15 @@ impl<'me, I: Interner> Solver<'me, I> {
                         subst: goal.trivial_substitution(self.program.interner()),
                         constraints: vec![],
                     };
+
                     debug!("applying coinductive semantics");
-                    return Ok(Solution::Unique(Canonical {
-                        value,
-                        binders: goal.canonical.binders,
-                    }));
+                    return Ok(Answer::new(
+                        Solution::Unique(Canonical {
+                            value,
+                            binders: goal.clone().canonical.binders,
+                        }),
+                        vec![goal],
+                    ));
```

Comment:
So I was to think a bit about this -- in particular, I think the recursive solver can handle this case differently than the SLG solver. Unlike the SLG solver, it's ok for intermediate queries in the recursive solver to not get cached (and/or get cached at the end, along with everything else). I have to review how we handle caching and minimums, but I suspect that we want to work towards a setup where the cache itself **never** has delayed goals, they are always ephemeral things that exist on the stack during solving.